### PR TITLE
Footer premium terms fix

### PR DESCRIPTION
**Pull request by ChinmayMhatre**
changes and fixes talked in #6637

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/6687"></a>

### CODE REVIEW

**Comment by ChinmayMhatre** [comment]
@eddiejaoude  Fixes done .

File: pages/premium.js [hunk 1]
```diff
@@ -55,29 +55,30 @@ export default function Premium() {
           />
         )}
 
-        <div className="bg-white py-24 sm:py-32">
+        <div className="bg-white dark:bg-primary-high py-24 sm:py-32">
           <div className="mx-auto max-w-7xl px-6 lg:px-8">
             <div className="mx-auto grid max-w-2xl grid-cols-1 gap-x-8 gap-y-16 sm:gap-y-20 lg:mx-0 lg:max-w-none lg:grid-cols-3">
               <div>
-                <h2 className="text-base font-semibold leading-7 text-indigo-600">
+                <h2 className="text-base font-semibold leading-7 text-secondary-medium dark:text-secondary-low   ">
```
Comment (eddiejaoude) [inline pages/premium.js:62]:
```suggestion
                <h2 className="text-base font-semibold leading-7 text-secondary-medium dark:text-secondary-low ">
```

File: tests/events.spec.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ test("Click on events profile in navbar navigates to events page", async ({
   page,
 }) => {
   await page.goto("/");
-  await page.getByRole("link", { name: 'Events', exact: true }).click();
+  await page.getByRole('navigation').getByRole('link', { name: 'Events' }).click();
```
Comment (eddiejaoude) [inline tests/events.spec.js:9]:
```suggestion
  await page.getByRole("navigation").getByRole("link", { name: "Events" }).click();
```

File: tests/search.spec.js [hunk 3]
```diff
@@ -9,7 +9,7 @@ test("Search has title", async ({ page }) => {
 
 test("Navigate to the Search page", async ({ page }) => {
   await page.goto("/");
-  await page.locator("a:visible", { hasText: "Search" }).click();
+  await page.getByRole('navigation').getByRole('link', { name: 'Search' }).click();
```
Comment (eddiejaoude) [inline tests/search.spec.js:12]:
```suggestion
  await page.getByRole("navigation").getByRole("link", { name: "Search" }).click();
```

File: tests/playground.spec.js [hunk 4]
```diff
@@ -10,7 +10,7 @@ test("Playground has title", async ({ page }) => {
 
 test("Navigate to Playground", async ({ page }) => {
   await page.goto("/");
-  await page.locator("a:visible", { hasText: "Playground" }).click();
+  await page.getByRole('link', { name: 'Playground', exact: true }).click();
```
Comment (eddiejaoude) [inline tests/playground.spec.js:13]:
```suggestion
  await page.getByRole("link", { name: "Playground", exact: true }).click();
```

**Review by eddiejaoude — APPROVED** [review]
Looks great 👍  thank you Chinmay!

I left some minor comments but they are not blockers

**Pull request merged**